terra_notebook_utils/blobstore/local.py

```diff
@@ -0,0 +1,117 @@
+import os
+import shutil
+from math import ceil
+from functools import wraps
+from typing import Dict, Generator, Optional
+
+from getm import default_chunk_size
+
+from terra_notebook_utils import blobstore
+
+
+def catch_blob_not_found(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except FileNotFoundError as ex:
+            raise blobstore.BlobNotFoundError(f"Could not find {self.url}") from ex
+    return wrapper
+
+class LocalBlobStore(blobstore.BlobStore):
+    def __init__(self, basepath: str):
+        self.bucket_name = basepath
+
+    def list(self, prefix: str="") -> Generator["LocalBlob", None, None]:
+        root = os.path.join(self.bucket_name, prefix)
+        if root.endswith(os.path.sep):
+            root = root[:-len(os.path.sep)]
+        for (dirpath, dirnames, filenames) in os.walk(root):
+            for filename in filenames:
+                relpath = os.path.relpath(os.path.join(dirpath, filename), self.bucket_name)
+                yield LocalBlob(self.bucket_name, relpath)
+
+    def blob(self, key: str) -> "LocalBlob":
+        return LocalBlob(self.bucket_name, key)
+
+class LocalBlob(blobstore.Blob):
+    def __init__(self, basepath: str, relpath: str):
```

Comment:
Is basepath always the same directory (`pwd`)?  Would it make it simpler to just accept one absolute (or relative) path?

Reply:
basepath can be any directory.

This mechanism is more complex than needed, but was pulled from working code so I don't want to make a lot of changes now. I'll consider simplifications in the future.